### PR TITLE
[PyOV] OVDict class - new return value from inference

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
- Add new class that improves return value from inference calls like `InferRequest.infer` or `CompiledModel.__call__`.
- Support keys of types: `ConstOutput`/`int`/`str`
- Inspired by "frozendict" approach.
- Example usage:
```python
In [12]: print(compiled_model)
<CompiledModel:
inputs[
<ConstOutput: names[data_0] shape[1,10] type: f32>
]
outputs[
<ConstOutput: names[my_output] shape[1,10] type: f32>
]>

In [13]: result = compiled_model(input_data)

In [14]: result[compiled_model.outputs[0]]
Out[14]: 
array([[0.43329707, 0.1623618 , 0.83914065, 0.91916376, 0.6939009 ,
        0.42150208, 0.4276818 , 0.42574805, 0.4555329 , 0.7696219 ]],
      dtype=float32)

In [15]: result[0]
Out[15]: 
array([[0.43329707, 0.1623618 , 0.83914065, 0.91916376, 0.6939009 ,
        0.42150208, 0.4276818 , 0.42574805, 0.4555329 , 0.7696219 ]],
      dtype=float32)

In [16]: result["my_output"]
Out[16]: 
array([[0.43329707, 0.1623618 , 0.83914065, 0.91916376, 0.6939009 ,
        0.42150208, 0.4276818 , 0.42574805, 0.4555329 , 0.7696219 ]],
      dtype=float32)

In [17]: res0, = compiled_model(input_data).to_tuple()

In [18]: res0
Out[18]: 
array([[0.43329707, 0.1623618 , 0.83914065, 0.91916376, 0.6939009 ,
        0.42150208, 0.4276818 , 0.42574805, 0.4555329 , 0.7696219 ]],
      dtype=float32)
```

### Tickets:
 - *97201*

### CODE REVIEW

File: src/bindings/python/tests/test_runtime/test_ovdict.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
```
Comment (ilya-lavrenov) [inline src/bindings/python/tests/test_runtime/test_ovdict.py:3]: should we also tests for InferRequest which show that tensors can be accessed using new approach?

Comment (jiwaszki) [reply src/bindings/python/tests/test_runtime/test_ovdict.py:3]: All of these "tensor-getters" are present since 2022.1 release.
Some tests:
https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_runtime/test_infer_request.py#L125
https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_runtime/test_infer_request.py#L202
There are no changes to `InferRequest` itself in this PR. Only affected part is the return value of specific infer functions. Thus new tests would not cover anything more.

@ilya-lavrenov Did I understand you correctly?

**Comment by jiwaszki** [comment]
Adding "do not merge", awaiting additional reviews (CC @akuporos @AlexKoff88 )

File: src/bindings/python/src/openvino/runtime/utils/data_helpers/wrappers.py [hunk 2]
```diff
@@ -20,3 +30,109 @@ def __init__(self, other: InferRequestBase) -> None:
         # Private memeber to store newly created shared memory data
         self._inputs_data = None
         super().__init__(other)
+
+
+class OVDict(Mapping):
+    """Custom OpenVINO dictionary with inference results.
+
+    This class is a dict-like object. It provides possibility to
+    address each element with three key types:
```
Comment (AlexKoff88) [inline src/bindings/python/src/openvino/runtime/utils/data_helpers/wrappers.py:39]: I would rephrase this like `...access data tensors with three...`

File: src/bindings/python/src/openvino/runtime/utils/data_helpers/wrappers.py [hunk 3]
```diff
@@ -20,3 +30,109 @@ def __init__(self, other: InferRequestBase) -> None:
...
+
+    This class follows `frozenset`/`tuple` concept of immutability.
+    It is prohibited to assign new items or edit them.
+
+    To revert to the previous behavior use `to_dict` method which
+    return shallow copy of underlaying dictionary.
+    Note: It removes addressing feature! New dictionary keeps
+          only `ConstOutput` keys.
+
+    If a tuple return value is needed, use `to_tuple` method which
```
Comment (AlexKoff88) [inline src/bindings/python/src/openvino/runtime/utils/data_helpers/wrappers.py:53]: `return` -> `returns`

**Review by AlexKoff88 — APPROVED** [review]
